src/hyperlink/_url.py

```diff
@@ -1410,24 +1504,26 @@ def to_text(self, with_password=False):
             password)."
 
         Args:
-            with_password (bool): Whether or not to include the
-               password in the URL text. Defaults to False.
+            with_password: Whether or not to include the password in the URL
+                text. Defaults to False.
 
         Returns:
-            str: The serialized textual representation of this URL,
-            such as ``u"http://example.com/some/path?some=query"``.
+            The serialized textual representation of this URL, such as
+                ``u"http://example.com/some/path?some=query"``.
 
         The natural counterpart to :class:`URL.from_text()`.
 
         .. _section 3.2.1: https://tools.ietf.org/html/rfc3986#section-3.2.1
         """
         scheme = self.scheme
         authority = self.authority(with_password)
-        path = _encode_path_parts(self.path,
-                                  rooted=self.rooted,
-                                  has_scheme=bool(scheme),
-                                  has_authority=bool(authority),
-                                  maximal=False)
+        path = "/".join(_encode_path_parts(
+            self.path,
+            rooted=self.rooted,
+            has_scheme=bool(scheme),
+            has_authority=bool(authority),
+            maximal=False
+        ))
```

Comment:
`_encode_path_parts` no longer returns a string by default (it returns a sequence of parts).